ngraph/core/src/op/squeeze.cpp

```diff
@@ -61,7 +82,7 @@ void op::Squeeze::pre_validate_and_infer_types()
         normalize_axes(this->description(), axes_constant->cast_vector<int64_t>(), data_rank);
 
     // Prepare set of unique axes marked to be removed from input data.
-    vector<bool> axes_to_squeeze(data_rank);
+    vector<bool> axes_to_squeeze(data_rank, false);
     if (axes_is_empty_constant)
     {
         auto data_shape = data.get_shape();
```

Comment:
We can extend this check for dynamic shapes with static rank. 
If there is no `compatible(1)` dim in the input shape, that means there is nothing to squeeze and we can pass input shape as output shape which preserves more information than fully dynamic shape.

Reply:
Yeah, this is a good point. I actually wanted to do something similar in the beginning, just did it the wrong way 👍